app/global/MainModule.java

```diff
@@ -84,6 +86,21 @@ protected void configure() {
     }
 
     @Singleton
+    @Provides
+    @SuppressFBWarnings("UPM_UNCALLED_PRIVATE_METHOD") // Invoked reflectively by Guice
+    private HealthProvider getHealthProvider(final Environment environment, final Configuration config) {
+        try {
+            final Class<? extends HealthProvider> clazz = ConfigurationHelper.getType(
+                    environment,
+                    config,
+                    "http.healthProvider");
+            return clazz.newInstance();
```

Comment:
You're going to need to support parameters on the object here.  You'll probably also want it dependency injected.  I recommend you take a look at what I did in Artemis: https://github.com/groupon/artemis/blob/master/app/global/ProdModule.java#L100-L120